acslib/ccure/base.py

```diff
@@ -1,22 +1,171 @@
 from typing import Optional
 
-from acslib.base import AccessControlSystem
-from acslib.ccure.connection import CcureConnection
+from acslib.base import AccessControlSystem, ACSRequestData, ACSRequestResponse, ACSRequestException
+from acslib.ccure.connection import CcureConnection, ACSRequestMethod
+from acslib.ccure.filters import CcureFilter, NFUZZ
 
 
 class CcureACS(AccessControlSystem):
-    """."""
+    """Base class for CCure API interactions"""
 
     def __init__(self, connection: Optional[CcureConnection] = None):
-        """."""
         super().__init__(connection=connection)
         if not self.connection:
             self.connection = CcureConnection()
         self.logger = self.connection.logger
         self.request_options = {}
-        self.search_filter = None
 
     @property
     def config(self):
         """."""
         return self.connection.config
+
+    def search(
+        self,
+        object_type: str,
+        search_filter: CcureFilter,
+        terms: Optional[list] = None,
+        page_size: Optional[int] = None,
+        page_number: int = 1,
+        search_options: dict = {},
+        where_clause: Optional[str] = None,
+    ):
+        """
+        Return CCure objects meeting the given criteria
+
+        object_type: full name of the CCure object type, such as those in ObjectType
+        search_filter: CcureFilter object specifying search query and display properties
+        terms: search terms used to filter objects. Leave empty to include everything in results
+        page_size: number of search results to include
+        page_number: the page of search results to display. The first page is page 1.
+        search_options: other options to include in the request_json. eg. "CountOnly"
+        where_clause: sql-style WHERE clause to search objects. overrides `terms` if included.
+        """
+        if page_size is None:
+            page_size = self.config.page_size
+        request_json = {
+            "TypeFullName": object_type,
+            "pageSize": page_size,
+            "pageNumber": page_number,
+            "DisplayProperties": search_filter.display_properties,
+            "WhereClause": where_clause or search_filter.filter(terms),
+        } | search_options
```

Comment:
`search_options` lets the user add fields to the `request_json` dict.
`page_size` and `page_number` are both just fields in the same `request_json` dict.
I made those two variables args in the search method because they might get used pretty often, but would it be cleaner/better to remove those args and just let the user add `"page_size"` and `"page_number"` to `search_options` if they want?

Reply:
I think it's fine to leave them in the arguments. What is the default if `page_size` is None?

Reply:
It defaults to whatever we have in the config file, 100 in our case

Reply:
We should add that to the docstring